fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/MatchCandidate.java

```diff
@@ -314,7 +316,7 @@ static Optional<MatchCandidate> fromPrimaryDefinition(@Nonnull final RecordMetaD
                                                           final boolean isReverse) {
         if (commonPrimaryKey != null) {
             final var availableRecordTypes = metaData.getRecordTypes().keySet();
-            final var baseRef = createBaseRef(metaData, availableRecordTypes, recordTypes);
+            final var baseRef = createBaseRef(metaData, availableRecordTypes, recordTypes, "");
```

Comment:
This feels hacky. How about defining a new class that can encapsulate these cases better:
1. match candidate side
    1. this is an index (there may even be other structures later as some indexes can be used in different ways, etc.)
    2. this is a primary access for the record type
2. query side
    1. no hinting for this access
    2. hinting to a set of indexes/and or primary

So maybe a new class called `AccessHints` which is handed in here that contains a set of `AccessHint`. `AccessHint` can say what it's hinting, i.e. a primary access or an index AND it can give the name if asked and even throw an exception if asked for the name of a primary access. `AccessHints` can encapsulate the check you currently do in `subsumedBy()` where you assign special meaning to the empty set.